### PR TITLE
Deploy to maven central

### PR DESCRIPTION
**Pull request by shjortConcordium**
## Purpose

Deploy to maven central as part of the release action.

## Changes

- Refactoring + added the maven central steps to release flow.
- Added plugins and profiles to poms to facilitate deployment. 
- Added info in parent pom
- Bumped versions to 6.1.0

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: concordium-android-sdk/.gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 /target
 /native/*
 /delombok
+
```
Comment (EmilLa1) [inline concordium-android-sdk/.gitignore:6]: nice :)

File: concordium-sdk/pom.xml [hunk 2]
```diff
@@ -5,16 +5,16 @@
     <modelVersion>4.0.0</modelVersion>
 
     <parent>
-        <groupId>com.concordium.sdk</groupId>
+    <groupId>com.concordium.sdk</groupId>
         <artifactId>concordium-sdk-base</artifactId>
-        <version>6.0.1-SNAPSHOT</version>
+        <version>6.0.2</version>
```
Comment (EmilLa1) [inline concordium-sdk/pom.xml:10]: Why not 6.1.0? New things are exposed in the API with this release right?

File: pom.xml [hunk 3]
```diff
@@ -5,12 +5,36 @@
 
     <groupId>com.concordium.sdk</groupId>
     <artifactId>concordium-sdk-base</artifactId>
-    <version>6.0.1-SNAPSHOT</version>
+    <version>6.0.2</version>
```
Comment (EmilLa1) [inline pom.xml:8]: Why not 6.1.0? New things are exposed in the API with this release right?

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -186,4 +176,39 @@ jobs:
...
+        mkdir -p concordium-android-sdk/target
+        && mv concordium-sdk/target/*-javadoc.jar concordium-android-sdk/target/concordium-android-sdk-javadoc.jar
+        && mv concordium-sdk/target/*-sources.jar concordium-android-sdk/target/concordium-android-sdk-sources.jar
+    - name: Publish android package to central maven repository
+      run: cd concordium-android-sdk && mvn --batch-mode deploy -Pdeploy
+      env:
+        MAVEN_USERNAME: ${{ secrets.SONATYPE_USR }}
+        MAVEN_PASSWORD: ${{ secrets.SONATYPE_PWD }}
+        GPG_PASSPHRASE: ${{secrets.SONATYPE_PHRASE }}
+      # Attach jar files to release (jar, jar-with-dependencies and javadoc-jar)
```
Comment (EmilLa1) [inline .github/workflows/release.yml:206]: Can you perhaps add the URLs of the deployed maven packages to the README. E.g. https://central.sonatype.com/artifact/com.concordium.sdk/concordium-android-sdk

Also, I believe we should only upload the "with-dependencies" artifact to the github release page, as the transitive dependencies cannot be resolved anyhow for the other one.

**Review by EmilLa1 — COMMENTED** [review]
Looks great! Minor questions and comments.

File: .github/workflows/release.yml (hunk 4 above)
Comment (orhoj) [reply .github/workflows/release.yml:206]: Do we even want to continue publishing to GitHub releases now that it will be available through Maven Central?

**Review by EmilLa1 — APPROVED** [review]

**Pull request merged**